app/waf/src/controllers/adc.controller.ts

```diff
@@ -751,6 +753,56 @@ export class AdcController extends BaseController {
     }
   }
 
+  private async isDOReady(adc: Adc): Promise<boolean> {
+    let cnct = adc.management.connection!;
+    let bigipMgr = await BigIpManager.instanlize({
+      username: cnct.username,
+      password: cnct.password,
+      ipAddr: cnct.ipAddress,
+      port: cnct.tcpPort,
+    });
+
+    try {
+      let resObj = await bigipMgr.getDOStatus();
+      let code = JSON.parse(resObj)['body'][0][0]['result']['status'];
+      return code === 'OK';
+    } catch {
+      console.log('DO URI has not been registered.');
+      return false;
+    }
+  }
+
+  private async doInstalling(adc: Adc): Promise<void> {
+    try {
+      this.logger.debug('start to install do');
+      //await this.serialize(adc, { status: AdcState.DOINSTALLING });
+      // check if do is already installed.
+      let cnct = adc.management.connection!;
+      let bigipMgr = await BigIpManager.instanlize({
+        username: cnct.username,
+        password: cnct.password,
+        ipAddr: cnct.ipAddress,
+        port: cnct.tcpPort,
+      });
+
+      if ((await this.isDOReady(adc)) === false) {
```

Comment:
Can we ignore the check, proceed the upload-install directly?

Reply:
I prefer to keep this logic this time and we may discuss this further.

Reply:
prefer to keep this logic this time and we could discuss this further.